File: backend/src/main/java/dev/laszlo/model/Choice.java
package dev.laszlo.model;

/**
 * Represents a choice that the user can make in the narrative.
 * Each choice leads to a different branch in the story and may switch the active character.
 */
public class Choice {
    private String id;              // Unique identifier (e.g., "choice_1", "ask_about_stars")
    private String label;           // Display text shown to user (e.g., "Ask about the constellation")
    private String nextSpeaker;     // Who responds next (e.g., "ilyra", "narrator")
    private String description;     // Optional tooltip/hover text for additional context

    // Default constructor
    public Choice() {
    }

    // Convenience constructor for basic choices
    public Choice(String id, String label, String nextSpeaker) {
        this.id = id;
        this.label = label;
        this.nextSpeaker = nextSpeaker;
    }

    // Full constructor
    public Choice(String id, String label, String nextSpeaker, String description) {
        this.id = id;
        this.label = label;
        this.nextSpeaker = nextSpeaker;
        this.description = description;
    }

    // Getters and Setters
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public String getNextSpeaker() {
        return nextSpeaker;
    }

    public void setNextSpeaker(String nextSpeaker) {
        this.nextSpeaker = nextSpeaker;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Choice{" +
                "id='" + id + '\'' +
                ", label='" + label + '\'' +
                ", nextSpeaker='" + nextSpeaker + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}

File: frontend/lib/models/choice.dart
// lib/models/choice.dart
// Represents a single choice that the user can make in the narrative

class Choice {
  final String id;              // "choice_1", "choice_2"
  final String label;           // "Ask about the stars"
  final String nextSpeaker;     // "ilyra" or "narrator"
  final String? description;    // Optional tooltip/hint

  Choice({
    required this.id,
    required this.label,
    required this.nextSpeaker,
    this.description,
  });

  /// Create Choice from JSON (from backend API or storage)
  /// Handles both 'id' (storage format) and 'choiceId' (legacy format)
  factory Choice.fromJson(Map<String, dynamic> json) {
    return Choice(
      id: (json['id'] ?? json['choiceId']) as String,
      label: json['label'] as String,
      nextSpeaker: json['nextSpeaker'] as String,
      description: json['description'] as String?,
    );
  }

  /// Convert Choice to JSON (for sending to backend)
  Map<String, dynamic> toJson() {
    return {
      'choiceId': id,
      'label': label,
      'nextSpeaker': nextSpeaker,
    };
  }

  /// Convert Choice to JSON for local storage persistence
  Map<String, dynamic> toStorageJson() {
    return {
      'id': id,
      'label': label,
      'nextSpeaker': nextSpeaker,
      'description': description,
    };
  }

  @override
  String toString() {
    return 'Choice(id: $id, label: $label, nextSpeaker: $nextSpeaker)';
  }
}